nexus/client.go

```diff
@@ -0,0 +1,543 @@
+package nexus
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"math"
+	"net/http"
+	"net/url"
+	"time"
+
+	"github.com/google/uuid"
+)
+
+// ClientOptions are options for creating a Client.
+type ClientOptions struct {
+	// Base URL of the service.
+	ServiceBaseURL string
+	// A function for making HTTP requests.
+	// Defaults to [http.DefaultClient.Do].
+	HTTPCaller func(*http.Request) (*http.Response, error)
+}
+
+// User-Agent header set on HTTP requests.
+const userAgent = "Nexus-go-sdk/" + version
+
+const headerUserAgent = "User-Agent"
+
+// Error indicating an empty ServiceBaseURL option was used to create a client when making a Nexus service request.
+var errEmptyServiceBaseURL = errors.New("empty serviceBaseURL")
+
+// Error indicating a non HTTP URL was used to create a [Client].
+var errInvalidURLScheme = errors.New("invalid URL scheme")
+
+var errEmptyOperationName = errors.New("empty operation name")
+
+var errEmptyOperationID = errors.New("empty operation ID")
+
+var errOperationWaitTimeout = errors.New("operation wait timeout")
```

Comment:
I still think you're predefining way too many things at the top of this file and this doesn't feel very Go like, but I'm not going to comment on it anymore